applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,242 +60,273 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
      */
-    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting, ETagFilter filter) {
+    public Mono<RestResponse<ConfigurationSetting>> addKeyValue(ConfigurationSetting configurationSetting) {
         Validator.validate(configurationSetting);
-        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters().withValue(configurationSetting.value())
+        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters()
+                .withValue(configurationSetting.value())
                 .withContentType(configurationSetting.contentType())
                 .withTags(configurationSetting.tags());
 
-        if (filter != null) {
-            return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, null, null);
+        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, null, getETagValue(ETAG_ANY));
     }
 
     /**
-     * Gets the ConfigurationSetting object for the specified key and KeyValueFilter2 parameters.
-     * @param key the key being retrieved
-     * @param filter options for the request
-     * @return ConfigurationSetting object
+     * Creates or updates a configuration value in the service.
+     * <p>
+     * If {@link ConfigurationSetting#etag()} is specified, the configuration value is updated if the current setting's
+     * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
+     * updated.
+     *
+     * @param configurationSetting The configuration setting to create or update.
+     * @return ConfigurationSetting that was created or updated.
+     * @throws com.microsoft.azure.v3.CloudException If the {@link ConfigurationSetting#etag()} was specified, is not
+     *                                               {@link ConfigurationClient#ETAG_ANY}, and the current configuration
+     *                                               value's etag does not match.
      */
-    public Mono<RestResponse<ConfigurationSetting>> getKeyValue(String key, KeyValueFilter filter) {
-        if (key == null || key.isEmpty()) {
-            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
-        }
-        if (filter != null) {
-            return service.getKeyValue(baseUri.toString(), key, filter.label(), filter.fields(),
-                    filter.acceptDateTime(), filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.getKeyValue(baseUri.toString(), key, null, null, null, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting) {
```

Comment:
Let's discuss the difference between set and add methods.

Reply:
This was something that the languages were trying to standardise behaviour for. [Standardization across languages](https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDeveloperExperience%2FShared%20Documents%2FService%20-%20Azconfig%2FAzure%20SDK%20-%20Service%20-%20Azconfig&wd=target%28Azconfig.one%7C50856656-5D71-2640-B240-11758DB2FB6E%2FStandarization%20across%20languages%7C79180E7A-F7B4-4221-8398-D440467E3782%2F%29)